flaskr_tests.py

```diff
@@ -0,0 +1,56 @@
+import os
+import flaskr
+import unittest
+import tempfile
+
+class FlaskrTestCase(unittest.TestCase):
+
+	def setUp(self):
+		self.db_fd, flaskr.app.config['DATABASE'] = tempfile.mkstemp()
+		flaskr.app.config['TESTING'] = True
+		self.app = flaskr.app.test_client()
+		with flaskr.app.app_context():
+			flaskr.init_db()
+
+	def tearDown(self):
+		os.close(self.db_fd)
+		os.unlink(flaskr.app.config['DATABASE'])
+
+	def test_empty_db(self):
+		rv = self.app.get('/')
+		assert b'No entries here so far' in rv.data
+
+	def login(self, username, password):
+		return self.app.post('/login', data=dict(
+			username=username,
+			password=password
+			), follow_redirects=True)
+
+	def logout(self):
+		return self.app.get('logout', follow_redirects=True)
+
+	def test_login_logout(self):
+		rv = self.login('admin','default')
+		assert b'You were logged in' in rv.data
+		
+		rv = self.logout()
+		assert b'You were logged out' in rv.data
+
+		rv = self.login('adminx','default')
+		assert b'Invalid username' in rv.data
+		
+		rv = self.login('admin','defaultx')
+		assert b'Invalid password' in rv.data
+
+	def test_messages(self):
+		self.login('admin', 'default')
+		rv = self.app.post('/add', data=dict(
+			title='<Hello>',
+			text='<strong>HTML</strong> allowed here'
+			), follow_redirects=True)
+		assert b'No entries here so far' not in rv.data
+		assert b'&lt;Hello&gt;' in rv.data
+		assert b'<strong>HTML</strong> allowed here' in rv.data
```

Comment:
I've redecided about `post_entry()` method, because it turns out to be the almost the same as `self.app.post()` :smile: , but anyway there're no tests in delete-entry branch, so if you'd like me to add them to that branch I'll be needing to somehow copy tests from edit-entry branch.